src/Catalog/CatalogItem.cs

```diff
@@ -37,5 +51,23 @@ public virtual IGraph CreateContentGraph(CatalogContext context)
         {
             return null;
         }
+
+        /// <summary>
+        /// Changes the host of the baseAddress with the customDomain
+        /// </summary>
+        /// <param name="baseAddress"></param>
+        /// <param name="customDomain"></param>
+        /// <returns></returns>
+        public static  Uri AliasesAddress(Uri baseAddress, string customDomain)
```

Comment:
It is used in the catalog content generation. IMO is to have it here or catalogwriter base. I added in the writerbase. If any strong preference let me know.